phys/module_physics_init.F

```diff
@@ -1115,37 +1120,82 @@ SUBROUTINE phy_init ( id, config_flags, DT, restart, zfull, zhalf,     &
       end do
     end do
 
-!..Fill initial starting values of radiative effective radii for
-!.. cloud water (2.51 microns), cloud ice (5.01 microns), and
-!.. snow (10.01 microns).
+!+---+-----------------------------------------------------------------+
+!..Fill initial starting values of radiative effective radii for cloud water,
+!.. cloud ice, and snow.  The very first timestep calls radiation before calling
+!.. microphysics, so it is better to supply something more realistic at the
+!.. onset rather than waiting until the 2nd radiation timestep.
+!+---+-----------------------------------------------------------------+
     if (has_reqc.ne.0) then
        do j=jts,jtf
           do k=kts,ktf
           do i=its,itf
-             re_cloud(i,k,j) = 2.51E-6
+             re_cloud(i,k,j) = 0.
+             IF (P_QC .GT. 1) THEN
+                Q_cld = MAX(0., moist(i,k,j,P_QC))
+             ENDIF
+             IF (P_QNC .GT. 1) THEN
+                N_cld = MAX(0., scalar(i,k,j,P_QNC))
+                IF (Q_cld .GT. 0) THEN
+                   re_cloud(i,k,j) = calc_effectRad_H20 (Q_cld, N_cld, inv_dens(i,k,j))
+                ENDIF
+             ELSE
+                IF ((XLAND(I,J)-1.5).GT.0.) THEN                         !--- Ocean
+                   re_cloud(i,k,j) = 9.5E-6
+                ELSE                                                     !--- Land
+                   re_cloud(i,k,j) = 5.5E-6
+                ENDIF
+             ENDIF
           end do
           end do
        end do
     endif
+
     if (has_reqi.ne.0) then
        do j=jts,jtf
           do k=kts,ktf
           do i=its,itf
-             re_ice(i,k,j) = 5.01E-6
+             re_ice(i,k,j) = 0.
+             IF (P_QI .GT. 1) THEN
+                Q_ice = MAX(0., moist(i,k,j,P_QI))
+             ENDIF
+             IF (P_QNI .GT. 1) THEN
+                N_ice = MAX(0., scalar(i,k,j,P_QNI))
+             ELSE
+                N_ice = 0.
+             ENDIF
+             IF (Q_ice .GT. 0) THEN
+                temper = theta(i,k,j) * (pressure(i,k,j)/p00)**(R_d/Cp)
```

Comment:
@gthompsnWRF @davegill Is this 'theta' always dry, or could it be 'theta_m'?

Reply:
@weiwangncar @gthompsnWRF 
Wei,
GREAT catch, I completely skipped past that issue.

Greg two points:
1. The theta array is dummy value associated the argument field grid%t_1. Depending on config_flags%use_theta_m (= 0 (dry), or = 1 (moist)), the grid%t_1 field can be either a moist or a dry theta.
2. The grid%t_1 field is a perturbation, not a full field.